File: jashing/src/main/java/com/github/avarabyeu/jashing/core/JashingEvent.java
package com.github.avarabyeu.jashing.core;

import java.time.Instant;

/**
 * Base Event for all Jashing events
 *
 * @author avarabyeu
 */
public class JashingEvent {
    private String id;
    private long updatedAt;

    /* Do not populate if you don't need dynamic title */
    private String title;

    public JashingEvent() {
        this.updatedAt = Instant.now().toEpochMilli();
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public long getUpdatedAt() {
        return updatedAt;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }
}

File: jashing/src/main/kotlin/com/github/avarabyeu/jashing/core/JashingEvent.kt
package com.github.avarabyeu.jashing.core

import com.github.avarabyeu.jashing.core.eventsource.annotation.EventId
import java.time.Instant

/**
 * Base Event for all Jashing events

 * @author avarabyeu
 */
open class JashingEvent {
    @EventId lateinit var id: String
    val updatedAt: Long

    /* Do not populate if you don't need dynamic title */
    var title: String? = null

    init {
        this.updatedAt = Instant.now().epochSecond
    }
}
